api/client/build.go

```diff
@@ -324,6 +312,24 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	return nil
 }
 
+// validateTag checks if the given image name can be resolved.
+func validateTag(rawRepo string) (string, error) {
```

Comment:
@tiborvass This function does nothing more than validation. On the server side the function is named `parseRepoAndTags`, which does the correction logic.
Since the server treats duplicated tags as valid input, I think there's no need to remove the duplicated tags on the client side.